nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp

```diff
@@ -73,8 +83,27 @@ inline BT::NodeStatus GoalUpdater::tick()
         last_goal_received_time.seconds(), goal_time.seconds());
     }
   }
-
   setOutput("output_goal", goal);
+
+  if (last_goals_received_.header.stamp != rclcpp::Time(0) && !last_goals_received_.poses.empty()) {
```

Comment:
I have a question about the intended behavior when `last_goal_received_.header.stamp == rclcpp::Time(0)`. When the topic-received goal has no timestamp, do we want to use that goal or reject it? Currently we don't use it but also don't show any warning. Looks like unintended behavior because if it has a timestamp but it's too old, we don't use it but we do warn

Reply:
If the timestamp isn't implemented for an incoming goal topic update, we skip it entirely. I think a log would be fantastic, yes!